src/go/rpk/pkg/tuners/gcp_disk_write_cache_tuner_test.go

```diff
@@ -0,0 +1,200 @@
+
+package tuners
+
+import (
+	"fmt"
+	"testing"
+	"vectorized/pkg/cloud/vendor"
+	"vectorized/pkg/tuners/disk"
+	"vectorized/pkg/tuners/executors"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/require"
+)
+
+type vendorMock struct {
+	init func() (vendor.InitializedVendor, error)
+}
+
+type currentVendor struct {
+	name string
+}
+
+func (v *vendorMock) Init() (vendor.InitializedVendor, error) {
+	return v.init()
+}
+
+func (v *vendorMock) Name() string {
+	return "none"
+}
+
+func (v *currentVendor) Name() string {
+	return v.name
+}
+
+func (v *currentVendor) VmType() (string, error) {
+	return "", nil
+}
+
+const devicePath = "/sys/devices/pci0000:00/0000:00:1d.0/0000:71:00.0/nvme/fake"
+
+func TestDeviceWriteCacheTuner_Tune(t *testing.T) {
+	// given
+	v := &vendorMock{
+		init: func() (vendor.InitializedVendor, error) {
+			return &currentVendor{
+				name: "gcp",
+			}, nil
+		},
+	}
+	deviceFeatures := &deviceFeaturesMock{
+		getWriteCacheFeatureFile: func(string) (string, error) {
+			return devicePath + "/queue/write_cache", nil
+		},
+		getWriteCache: func(string) (string, error) {
+			return "write back", nil
+		},
+	}
+	fs := afero.NewMemMapFs()
+	fs.MkdirAll(devicePath+"/queue", 0644)
+	tuner := NewDeviceGcpWriteCacheTuner(fs, "fake", deviceFeatures, v, executors.NewDirectExecutor())
+	// when
+	tuner.Tune()
+	// then
+	setValue, _ := afero.ReadFile(fs, devicePath+"/queue/write_cache")
+	require.Equal(t, "write through", string(setValue))
+}
+
+func TestDeviceWriteCacheTuner_IsSupported_Should_return_true(t *testing.T) {
+	// given
+	v := &vendorMock{
+		init: func() (vendor.InitializedVendor, error) {
+			return &currentVendor{
+				name: "gcp",
+			}, nil
+		},
+	}
+	deviceFeatures := &deviceFeaturesMock{
+		getWriteCacheFeatureFile: func(string) (string, error) {
+			return devicePath + "/queue/write_cache", nil
+		},
+		getWriteCache: func(string) (string, error) {
+			return "write back", nil
+		},
+	}
+	fs := afero.NewMemMapFs()
+	fs.MkdirAll(devicePath+"/queue", 0644)
+	tuner := NewDeviceGcpWriteCacheTuner(fs, "fake", deviceFeatures, v, executors.NewDirectExecutor())
+	// when
+	supported, _ := tuner.CheckIfSupported()
+	// then
+	require.True(t, supported)
+}
+
+func TestDeviceWriteCacheTuner_IsSupported_should_return_false_on_aws(
+	t *testing.T,
+) {
+	// given
+	v := &vendorMock{
+		init: func() (vendor.InitializedVendor, error) {
+			return &currentVendor{
+				name: "aws",
+			}, nil
+		},
+	}
+	deviceFeatures := &deviceFeaturesMock{
+		getWriteCacheFeatureFile: func(string) (string, error) {
+			return devicePath + "/queue/write_cache", nil
+		},
+		getWriteCache: func(string) (string, error) {
+			return "write back", nil
+		},
+	}
+	fs := afero.NewMemMapFs()
+	fs.MkdirAll(devicePath+"/queue", 0644)
+	tuner := NewDeviceGcpWriteCacheTuner(fs, "fake", deviceFeatures, v, executors.NewDirectExecutor())
+	// when
+	supported, _ := tuner.CheckIfSupported()
+	// then
+	require.False(t, supported)
```

Comment:
Can you make these ^ table tests too? They make it so much easier to add new tests, say when we add Azure & other providers.